languages/python/sqlalchemy-oso/sqlalchemy_oso/session.py

```diff
@@ -1,8 +1,10 @@
 """SQLAlchemy session classes and factories for oso."""
-from typing import Any, Callable, Dict, Optional, Type
+from typing import Any, Callable, Dict, Optional, Type, Union
+import logging
 
 from sqlalchemy import event, inspect
 from sqlalchemy.orm import sessionmaker, Session
+from sqlalchemy.ext.asyncio import AsyncSession, async_scoped_session
```

Comment:
I think we'll want to move this import into a method so that this file won't cause errors using pre-1.4 sqlalchemy. @dhatch it looks like you did something to this effect a little while ago when you were adjusting the way `joinedload`s get authorized. Do you have any pointers here?